misc/python/materialize/checks/mz_version.py

```diff
@@ -0,0 +1,64 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import json
+import subprocess
+from functools import total_ordering
+from typing import Any
+
+
+@total_ordering
+class MzVersion:
+    """Version of Materialize based on mz_version() string."""
+
+    def __init__(self, version: str) -> None:
+        if not version[0] == "v":
+            raise ValueError(f"Invalid mz version string: {version}")
+        if " " in version:
+            version, git_hash = version.split(" ")
+            if not git_hash[0] == "(" or not git_hash[-1] == ")":
+                raise ValueError(f"Invalid mz version string: {version}")
+            self._hash = git_hash  # Currently ignored
+        self._dev = version.endswith("-dev")  # Currently ignored
+        version = version.removesuffix("-dev")
+        self._version = tuple(int(part) for part in version[1:].split("."))
```

Comment:
You can totally offload the correctness checking and the comparison to the `semver` module. It will do the splitting internally, as well as removing the `-dev` suffix. It also allows comparisons between semver objects and strings and whatnot, plus the constructor has a bunch of complex-beyond-comprehension validation regexps.

Reply:
Oh, my bad. When you said to use semver, I just thought you meant semantic versioning itself, but you meant the module. Will do.